### PR TITLE
Update Rust edition to 2021

### PR DESCRIPTION
**Pull request by Bargsteen**
## Purpose

Update the Rust edition to 2021 as it is a requirement when using the integration testing library.

## Changes

- Update editions to 2021
- In 2021, prefixed identifiers and literals are reserved syntax, so I used `cargo fix --edition` for the migration. More info at: https://doc.rust-lang.org/edition-guide/rust-2021/reserving-syntax.html

## Checklist

- [x] My code follows the style of this project.
- [x] The code compiles without warnings.
- [x] I have performed a self-review of the changes.
- [x] I have documented my code, in particular the intent of the
      hard-to-understand areas.
- [x] (If necessary) I have updated the CHANGELOG.

### CODE REVIEW

File: concordium-std-derive/src/lib.rs [hunk 1]
```diff
@@ -2417,17 +2417,17 @@ fn impl_state_clone(ast: &syn::DeriveInput) -> syn::Result<TokenStream> {
...
-                        quote!(#data_name::#variant_ident#field_names),
+                        quote!(#data_name::#variant_ident #field_names),
                         proc_macro2::TokenStream::new(),
                     ),
                 };
                 let variant_ident = &variant.ident;
 
                 matches_tokens.extend(quote! {
-                    #data_name::#variant_ident#pattern => {
+                    #data_name::#variant_ident #pattern => {
```
Comment (abizjak) [inline concordium-std-derive/src/lib.rs:2430]: Where do these changes come from?

File: examples/piggy-bank/part2/Cargo.toml [hunk 2]
```diff
@@ -19,8 +19,10 @@ wee_alloc = ["concordium-std/wee_alloc"]
 
 [dependencies.concordium-std]
 version = "6"
-path = "../../../concordium-std"
 default-features = false
 
+[dev-dependencies]
+concordium-smart-contract-testing = { path = "../../../../concordium-smart-contract-tools/concordium-smart-contract-testing", version = "1.0" }
```
Comment (abizjak) [inline examples/piggy-bank/part2/Cargo.toml:25]: I don't think this belongs here.

**Review by abizjak — APPROVED** [review]
Looks fine, but some jobs are failing due to trivial issues.

File: concordium-std-derive/src/lib.rs (hunk 1 above)
Comment (Bargsteen) [reply concordium-std-derive/src/lib.rs:2430]: `cargo fix --edition`. It is related to the new reserved syntax mentioned above.

File: examples/piggy-bank/part2/Cargo.toml (hunk 2 above)
Comment (Bargsteen) [reply examples/piggy-bank/part2/Cargo.toml:25]: No, certainly not 🙃 I'll fix that.

**Review by abizjak — APPROVED** [review]

**Pull request merged**